src/server/handlers/launchpad.js

```diff
@@ -340,6 +340,47 @@ export const internalFindSnap = async (repositoryUrl) => {
   });
 };
 
+const internalFindSnapsByPrefix = (urlPrefix) => {
+  const username = conf.get('LP_API_USERNAME');
+
+  return getLaunchpad().named_get('/+snaps', 'findByURLPrefix', {
```

Comment:
Yes, we need to memcache this, as it's potentially somewhat expensive to compute and we'll be calling it quite a bit.  However, we need to make sure to clear relevant cached values when we make changes.